sdk/azcore/example_test.go

```diff
@@ -58,3 +63,25 @@ func ExampleNullValue() {
 	// Output:
 	// {"count":null}
 }
+
+func ExampleHTTPResponse() {
+	pipeline := runtime.NewPipeline(http.DefaultClient)
+	req, err := runtime.NewRequest(context.Background(), "POST", "https://fakecontainerregisty.azurecr.io/acr/v1/nonexisteng/_tags")
+	if err != nil {
+		panic(err)
+	}
+	resp, err := pipeline.Do(req)
+	var httpErr azcore.HTTPResponse
+	if errors.As(err, &httpErr) {
+		// Handle Error
+		if httpErr.RawResponse().StatusCode == http.StatusNotFound {
+			fmt.Printf("Repository could not be found: %v", httpErr.RawResponse())
+		} else if httpErr.RawResponse().StatusCode == http.StatusForbidden {
+			fmt.Printf("You do not have permission to access this repository: %v", httpErr.RawResponse())
+		} else {
+			// ...
+		}
+	}
+	// Do something with response
+	fmt.Println(ioutil.ReadAll(resp.Body))
```

Comment:
Should we check that resp is not nil? Ie, are azcore.HTTPResponse's the only errors that can come back here?

Reply:
So `HTTPResponse` is the interface used to get at the underlying `*http.Response` from an error returned by SDK code due to a failure HTTP status code.  In the context of this example, `Do()` will never return an `HTTPResponse` error because there's no HTTP status code to predicate success/failure.